### PR TITLE
Trying to merge the branches 

### PR DESCRIPTION
**Pull request by Duologic**
* `pre-django-1.8`: currently used internally
* `master`: version used by the outside world?
* `pre-django-1.8-merge-master`: changes from `master` merged into `pre-django-1.8`

This pull request is to merge `pre-django-1.8-merge-master` into `master`.

### CODE REVIEW

**Comment by Duologic** [comment]
Rebased this branch on master.

File: locking/tests.py [hunk 1]
```diff
@@ -122,15 +121,16 @@ def test_context_manager(self):
...
 class CleanExpiredLocksTest(TestCase):
+    """Tests correct functioning of the task that cleans expired locks."""
     def setUp(self):
-        self.user = User.objects.create(username='CleanExpiredLocks test')
+        self.user = User.objects.create(username='hellofoo')
 
-    def test_clean_locks(self):
+    def test_clean(self):
+        """Make expired lock, ensure the management command cleans it up."""
         with freeze_time("2015-01-01 10:00"):
```
Comment (RoelandMatthijssens) [inline locking/tests.py]: Are we intentionally still using freezegun in this repo? instead of faketime

**Comment by Duologic** [comment]
`pre-django-1.8` is now in production.

You can make a diff between this branch and `pre-django-1.8`.

**Comment by Duologic** [comment]
Also, look what I found: https://github.com/vikingco/django-db-locking/commit/b8f468920e5e048f20ee19af4c8d0b7638642e31#diff-53e2cf494c60565890e80f4274c8d182R46

Seems the atomic blocks have been discussed before... 💃

File: locking/models.py [hunk 2]
```diff
@@ -50,21 +50,30 @@ def acquire_lock(self, obj=None, max_age=None, lock_name=''):
...
-                    # Create a new lock to provide a new id for renewal.
-                    # This ensures the owner of the previous lock doesn't
-                    # remain in possession of the active lock id.
-                    lock.release()
-                    lock = self.create(locked_object=lock_name, max_age=max_age)
-                else:
-                    raise AlreadyLocked()
-        except IntegrityError:
-            raise AlreadyLocked()
+        with transaction.atomic():
```
Comment (KMahoney) [inline locking/models.py]: As I mentioned in the the PR from a couple of years ago, I don't think this atomic block provides much value. From what I can tell it's already in production, so if there haven't been any issues (such as deadlocks) I'm happy to leave as-is.

**Pull request merged**